tools/distpartitioning/data_shuffle.py

```diff
@@ -12,82 +12,88 @@
 from datetime import timedelta
 from dataset_utils import get_dataset
 from utils import read_partitions_file, read_json, get_node_types, \
-                    augment_node_data, augment_edge_data, get_ntypes_map, \
+                    augment_edge_data, get_ntypes_map, \
                     write_dgl_objects, write_metadata_json
 from gloo_wrapper import alltoall_cpu_object_lst, alltoallv_cpu, \
                     alltoall_cpu, allgather_sizes, gather_metadata_json
 from globalids import assign_shuffle_global_nids_nodes, \
                     assign_shuffle_global_nids_edges, get_shuffle_global_nids_edges
 from convert_partition import create_dgl_object, create_metadata_json, validateDGLObjects
 
-def exchange_node_data(rank, world_size, node_data):
-    """
-    Exchange node_data among the processes in the world
-    Prepare the list of slices targeting each of the process and
-    trigger alltoallv_cpu for the message exchange.
+def gen_node_data(rank, world_size, node_part_ids, ntid_cnt_map, ntid_ntype_map, nid_schema_map):
+    '''
+    For this data processing pipeline, reading node files is not needed. All the needed information about
+    the nodes can be found in the metadata json file. This function generates the nodes owned by a given
+    process, using metis partitions. 
 
-    Parameters:
+    Parameters: 
     -----------
     rank : int
-        rank of the current process
+        rank of the process
     world_size : int
-        total no. of participating processes
-    node_data : dictionary
-        nodes data dictionary with keys as column names and values as
-        columns from the nodes csv file
+        total no. of processes
+    node_part_ids : 
+        numpy array, whose length is same as no. of nodes in the graph. Index in this array is the global_nid
+        and value is the partition-id which owns the node
+    ntid_cnt_map : 
+        a dictionary where keys are node_type ids and values are total nodes which are of that node_type
+    ntid_ntype_map : 
+        a dictionary where keys are node_type ids and values are node_type names
+    nid_schema_map: 
+        a dictionary, which is extracted from the input graph metadata json file for node information, 
+        key is `nid`
 
     Returns:
     --------
     dictionary : 
-        the input argument, node_data, is updated with the node data received by other processes
-        in the world.
-    """
+        dictionary where keys are column names and values are numpy arrays, these arrays are generated by 
+        using information present in the metadata json file
+
+    '''
+    '''
+    ntypes_nid_map -> ntype_name -> str(ntype_id)
+    ntypes_id_count -> str(ntype_id) -> total node count
+    '''
+
+    local_node_data = {}
+    gnid_start = 0
+    gnid_end = 0
+    for ntypeid in range(len(ntid_cnt_map)):
+        ntype_name = ntid_ntype_map[str(ntypeid)]
```

Comment:
why is node type id a string?

Reply:
This is how it is stored in the dictionaries. In the json object, for an object to be serializable it is converted into string form. This is the reason, all the keys which are integers are in string forms when json object is created.

Reply:
why do we store node type IDs in the JSON file?